Infrastructure.DataAccess/Interceptors/EFEntityInterceptor.cs

```diff
@@ -0,0 +1,150 @@
+
+        public EFEntityInterceptor(
+            Factory<IOperationClock> operationClock,
+            Factory<Maybe<ActiveUserIdContext>> userContext,
+            Factory<IFallbackUserResolver> fallbackUserResolver)
+        {
+            _operationClock = operationClock;
+            _userContext = userContext;
+            _fallbackUserResolver = fallbackUserResolver;
+        }
+
+        public void TreeCreated(DbCommandTreeInterceptionContext interceptionContext)
+        {
+            if (ShouldHandle(interceptionContext))
+            {
+                switch (interceptionContext.OriginalResult)
+                {
+                    case DbInsertCommandTree insertCommand:
+                        interceptionContext.Result = HandleInsertCommand(insertCommand);
+                        return;
+                    case DbUpdateCommandTree updateCommand:
+                        interceptionContext.Result = HandleUpdateCommand(updateCommand);
+                        return;
+                }
+            }
+        }
+
+        private static bool ShouldHandle(DbCommandTreeInterceptionContext interceptionContext)
+        {
+            return interceptionContext.OriginalResult.DataSpace == DataSpace.SSpace;
+        }
+
+        private DbCommandTree HandleInsertCommand(DbInsertCommandTree insertCommand)
+        {
+            var userId = GetActiveUserId();
+            var now = _operationClock().Now;
+
+            var updates = new List<KeyValuePair<Predicate<DbSetClause>, DbExpression>>
+            {
+                new KeyValuePair<Predicate<DbSetClause>, DbExpression>(clause=>MatchPropertyName(clause,ObjectOwnerIdColumnName), userId),
+                new KeyValuePair<Predicate<DbSetClause>, DbExpression>(clause=>MatchPropertyName(clause,LastChangedByUserIdColumnName), userId),
+                new KeyValuePair<Predicate<DbSetClause>, DbExpression>(clause=>MatchPropertyName(clause,LastChangedColumnName), DbExpression.FromDateTime(now))
+            };
+
+            var setClauses = insertCommand.SetClauses
+                .Select(clause => ApplyUpdates(clause, updates))
+                .ToList();
+
+            return new DbInsertCommandTree(
+                insertCommand.MetadataWorkspace,
+                insertCommand.DataSpace,
+                insertCommand.Target,
+                setClauses.AsReadOnly(),
+                insertCommand.Returning);
+        }
+
+        private DbCommandTree HandleUpdateCommand(DbUpdateCommandTree updateCommand)
+        {
+            var userId = GetActiveUserId();
+            var now = _operationClock().Now;
+
+            var updates = new List<KeyValuePair<Predicate<DbSetClause>, DbExpression>>
+            {
+                new KeyValuePair<Predicate<DbSetClause>, DbExpression>(clause=>MatchPropertyName(clause,ObjectOwnerIdColumnName) && MatchNull(clause), userId), //Some EF updates end up in this e.g. changing an owned child on a parent
```

Comment:
@JacobMalling OBS på denne. Den er lidt speciel. Det sker f.eks. hvor man skifter exhibit ud med en ny c# instans. Her laver den først en update hvor den afkobler den gamle (og hvor man så rammer denne case) og en anden INSERT med den nye instans der patches af HandleCreated... Det er altså lidt en edge case, men vores tests fangede den.

Reply:
Okay så her forsøgte den at ændre på child når den opdaterede parent?

Reply:
Nej problemet er at selvom "child" er et "child" i C# verdenen så er den jo sin egen tabel i SQL og når man laver de der udskiftningstricks, så kommer der nogle gange flere "validerings-loops" indtil den beslutter sig.. det er i hvert fald min teori for i databasen ser det korrekt ud bagefter :-)